stories/__tests__/SQFormButton.stories.test.js

```diff
@@ -0,0 +1,161 @@
+import userEvent from '@testing-library/user-event';
+import {composeStories} from '@storybook/testing-react';
+import * as stories from '../SQFormButton.stories';
+
+const {
+  Default: SQFormButton,
+  WithTestField: SQFormButtonWithField
+} = composeStories(stories);
+
+describe('Default Tests', () => {
+  it('should render a button with text', () => {
+    render(<SQFormButton />);
+
+    const submitButton = screen.getByRole('button', {name: /form submission/i});
+
+    expect(submitButton).toBeInTheDocument();
+    expect(submitButton).toHaveTextContent('Submit');
+  });
+
+  it('should render a reset button given the type', () => {
+    render(<SQFormButton type="reset" />);
+
+    const resetButton = screen.getByRole('button', {name: /form submission/i});
+
+    expect(resetButton).toBeInTheDocument();
+    expect(resetButton).toHaveAttribute('type', 'reset');
+  });
+
+  it('should call a function when clicked', () => {
+    const onClickSpy = jest.fn();
+    render(<SQFormButton onClick={onClickSpy} />);
+
+    const submitButton = screen.getByRole('button', {name: /form submission/i});
+
+    userEvent.click(submitButton);
+
+    expect(onClickSpy).toHaveBeenCalled();
+  });
+
+  it('should show as disabled', () => {
+    render(<SQFormButton isDisabled={true} />);
+
+    const submitButton = screen.getByRole('button', {name: /form submission/i});
+
+    expect(submitButton).toBeDisabled();
+  });
+
+  it('should not fire onClick when disabled', () => {
+    const onClickSpy = jest.fn();
+    render(<SQFormButton isDisabled={true} onClick={onClickSpy} />);
+
+    const submitButton = screen.getByRole('button', {name: /form submission/i});
+
+    userEvent.click(submitButton);
+
+    expect(onClickSpy).not.toHaveBeenCalled();
+  });
+});
+
+describe('Button with Test Field Tests', () => {
+  it('should render a button and a text field', () => {
+    render(<SQFormButtonWithField />);
+
+    const testField = screen.getByLabelText(/test field/i);
+
+    expect(testField).toBeInTheDocument();
+
+    const submitButton = screen.getByRole('button', {name: /form submission/i});
+
+    expect(submitButton).toBeInTheDocument();
+    expect(submitButton).toHaveTextContent('Submit');
+  });
+
+  it('should disable when shouldRequireFieldUpdates is true', () => {
+    render(<SQFormButtonWithField shouldRequireFieldUpdates={true} />);
+
+    const submitButton = screen.getByRole('button', {name: /form submission/i});
+
+    expect(submitButton).toBeDisabled();
+  });
+
+  it('should enable when text field is filled out', () => {
+    render(<SQFormButtonWithField shouldRequireFieldUpdates={true} />);
+
+    const submitButton = screen.getByRole('button', {name: /form submission/i});
+    expect(submitButton).toBeDisabled();
+
+    const testField = screen.getByLabelText(/test field/i);
+
+    userEvent.type(testField, 'Hello');
+
+    expect(submitButton).toBeEnabled();
+  });
+
+  it('should submit form when submit button clicked', () => {
```

Comment:
I'm wondering if this test is really necessary here since the onClick has already been tested in the previous suite and testing the form submission has more to do with testing SQForm rather than the button

Reply:
I think you're right. This test isn't even really testing the form submit like the name suggests. I'll remove it.